sasmodels/sesans.py

```diff
@@ -58,23 +58,24 @@ def apply(self, Iq):
 
     def _set_hankel(self, SElength, lam, zaccept, Rmax):
         # type: (np.ndarray, float, float) -> None
-        # Force float32 arrays, otherwise run into memory problems on some machines
-        SElength = np.asarray(SElength, dtype='float32')
-
-        #Rmax = #value in text box somewhere in FitPage?
+        SElength = np.asarray(SElength)
         q_max = 2*pi / (SElength[1] - SElength[0])
         q_min = 0.1 * 2*pi / (np.size(SElength) * SElength[-1])
-        q = np.arange(q_min, q_max, q_min, dtype='float32')
-        dq = q_min
+        q = np.exp(np.arange(np.log(q_min), np.log(q_max), np.log(1.0003)))
+
+        dq = q[1:]-q[:-1]
+        dq = np.hstack([[dq[0]], dq])
 
-        H0 = np.float32(dq/(2*pi)) * q
+        H0 = dq/(2*pi) * q
 
-        repq = np.tile(q, (SElength.size, 1)).T
-        repSE = np.tile(SElength, (q.size, 1))
-        H = np.float32(dq/(2*pi)) * j0(repSE*repq) * repq
+        H = np.outer(q, SElength)
+        j0(H, out=H)
+        H *= (dq * q).reshape((-1, 1))
+        H /= 2*pi
```

Comment:
Can you do `H *= (dq * q / (2*pi)).reshape((-1,1)`?  This saves `(len(SElength)-1) * len(q)` operations.

Reply:
Good catch